### PR TITLE
New Endpoints

### PR DESCRIPTION
**Pull request by Kathund**
## Changes
**Most code is taken from #358 allowing it to be closed**

<ins>**[BREAKING] grouping into .skyblock for sb endpoints**</ins>

Adds new endpoints that are in the api but not supported in reborn

<details>
<summary>Screenshots</summary>

</details>

<details>
<summary>Checkboxes</summary>

- [x] I've added new features. (methods or parameters)
- [x] I've added jsdoc and typings.
- [ ] I've fixed bug. (_optional_ you can mention a issue if there is one)
- [ ] I've corrected the spelling in README, documentation, etc.
- [x] I've tested my code. (`npm run tests`)
- [x] I've check for issues. (`npm run eslint`)
- [x] I've fixed my formatting. (`npm run prettier`)

</details>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,4 +45,5 @@ testing/
 .circleci/
 changelog.md
 master.json
-.env
\ No newline at end of file
+.env
+.vscode/settings.json
```
Comment (ChiefChippy2) [inline .gitignore]:
```suggestion
.vscode/settings.json

```

File: src/API/skyblock/getAuction.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  let parameters = '';
+  if (type === 'PROFILE') {
+    parameters = `profile=${query}`;
+  } else if (type === 'PLAYER') {
+    query = await toUuid(query);
+    parameters = `player=${query}`;
+  } else if (type === 'AUCTION') {
+    parameters = `uuid=${query}`;
+  } else throw new Error(Errors.BAD_AUCTION_FILTER);
+  const res = await this._makeRequest(`/skyblock/auction?${parameters}`);
```
Comment (ChiefChippy2) [inline src/API/skyblock/getAuction.js]: imma kms if i wrote this

```suggestion
  let filter = '';
  if (type === 'PROFILE') {
    filter = 'profile';
  } else if (type === 'PLAYER') {
    query = await toUuid(query);
    filter = 'player';
  } else if (type === 'AUCTION') {
    filter = 'uuid';
  } else throw new Error(Errors.BAD_AUCTION_FILTER);
  const res = await this._makeRequest(`/skyblock/auction?${filter}=${query}`);
```

File: src/Client.js [hunk 3]
```diff
@@ -323,6 +338,27 @@ class Client extends EventEmitter {
    *   console.log(products[0].productId); // INK_SACK:3
    * })
    * .catch(console.log);
+   */ /**
+   * Allows you to get bingo data
+   * @method
+   * @name Client#getSkyblockBingo
```
Comment (ChiefChippy2) [inline src/Client.js]:
```suggestion
   * @name Client.skyblock#getBingo
```
(that's clearly not how you do it is it)

File: src/structures/SkyBlock/Static/Bingo.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/**
```
Comment (ChiefChippy2) [inline src/structures/SkyBlock/Static/Bingo.js:1]: i assume i wrote this

**Review by ChiefChippy2 — CHANGES_REQUESTED** [review]

File: src/structures/SkyBlock/Static/Bingo.js (hunk 4 above)
Comment (Kathund) [reply src/structures/SkyBlock/Static/Bingo.js:1]: yes?

**Comment by Kathund** [comment]
There is no current open mayor election so tests will fail

**Comment by Kathund** [comment]
@ChiefChippy2

**Comment by Kathund** [comment]
@ChiefChippy2

**Comment by ChiefChippy2** [comment]
will review on weekends

**Review by ChiefChippy2 — APPROVED** [review]

**Pull request merged**